x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -232,19 +238,76 @@ private static void validateField(Map<?, ?> properties, String fieldName, boolea
+                List<String> types = response.mappings()
+                    .values()
+                    .stream()
+                    .map(Map::values)
+                    .flatMap(Collection::stream)
+                    .map(toFieldMappingType(policy.getMatchField()))
+                    .distinct()
+                    .collect(Collectors.toList());
+                if (types.isEmpty()) {
+                    resultListener.onFailure(
+                        new ElasticsearchException(
+                            "No mapping type found for match field '{}' - indices({})",
+                            policy.getMatchField(),
+                            Strings.collectionToCommaDelimitedString(policy.getIndices())
+                        )
+                    );
+                } else if (types.size() > 1) {
+                    resultListener.onFailure(
+                        new ElasticsearchException(
+                            "Multiple distinct mapping types for match field '{}' - indices({})  types({})",
+                            policy.getMatchField(),
+                            Strings.collectionToCommaDelimitedString(policy.getIndices()),
+                            Strings.collectionToCommaDelimitedString(types)
+                        )
+                    );
+                } else {
+                    createEnrichMappingBuilder(
+                        (builder) -> builder.field("type", resolveMatchType(types.get(0))).field("doc_values", false),
+                        resultListener
+                    );
+                }
+            }));
             // No need to also configure index_options, because keyword type defaults to 'docs'.
         } else if (EnrichPolicy.GEO_MATCH_TYPE.equals(policy.getType())) {
-            matchFieldMapping = (builder) -> builder.field("type", "geo_shape");
+            createEnrichMappingBuilder((builder) -> builder.field("type", "geo_shape"), resultListener);
         } else {
-            throw new ElasticsearchException("Unrecognized enrich policy type [{}]", policy.getType());
+            resultListener.onFailure(new ElasticsearchException("Unrecognized enrich policy type [{}]", policy.getType()));
         }
+    }
+
+    @SuppressWarnings("unchecked")
+    private Function<GetFieldMappingsResponse.FieldMappingMetadata, String> toFieldMappingType(String field) {
+        return data -> ((Map<String, String>) data.sourceAsMap().getOrDefault(field, Map.of("type", "unknown"))).get("type");
+    }
 
+    private String resolveMatchType(String type) {
+        switch (type) {
+            case "integer_range":
+            case "float_range":
+            case "long_range":
+            case "double_range":
+            case "date_range":
+            case "ip_range":
+                return type;
+            case "text": // exact matching only ?
```

Comment:
I think this should be exact matching only for sure.

Reply:
Think this a moot point, now as if we did support fancier text matching it would be a different policy. Removed case